src/decorators/Subscription.ts

```diff
@@ -19,15 +20,18 @@ export function Subscription(
   maybeOptions?: SubscriptionOptions,
 ): MethodDecorator {
   const { options, returnTypeFunc } = getTypeDecoratorParams(returnTypeFuncOrOptions, maybeOptions);
-  const topics = ([] as string[]).concat(options.topics || []);
+  // const topics = ([] as string[]).concat(options.topics || []);
   return (prototype, methodName) => {
     const handler = getHandlerInfo(prototype, methodName, returnTypeFunc, options);
-    if (topics.length === 0) {
+    if (!options.topics) {
+      throw new MissingSubscriptionTopicsError(handler.target, handler.methodName);
+    }
+    if (Array.isArray(options.topics) && options.topics.length === 0) {
```

Comment:
We need this check also in the result of calling the `topics` function in `pubSubIterator` - string is checked by TS, but `string[]` is not